server/services/userService.ts

```diff
@@ -22,6 +25,25 @@ export default class UserService {
     return { ...user, caseloads, displayName: StringUtils.convertToTitleCase(user.name) }
   }
 
+  async getUserFromUsername(token: Express.User['token'], username: User['username']): Promise<User> {
```

Comment:
Should probably be used by `getCurrentUserWithDetails` too, rather than calling the client directly in there.